### PR TITLE
feat(batch): enable parallel table scan

### PR DESCRIPTION
**Pull request by xxchan**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?
We will not maintain the order of select from an MV with ORDER BY clause now. https://github.com/singularity-data/risingwave/pull/3668#issuecomment-1175217821

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* SQL commands, functions, and operators

### Release note

Select from an MV with ORDER BY clause may return unordered results. 

## Refer to a related PR or issue link (optional)
close https://github.com/singularity-data/risingwave/issues/3583
close https://github.com/singularity-data/risingwave/issues/3237

### CODE REVIEW

File: src/frontend/src/scheduler/plan_fragmenter.rs [hunk 1]
```diff
@@ -394,11 +394,7 @@ impl BatchPlanFragmenter {
                 if let Some(scan_node) = node.as_batch_seq_scan() {
                     let table_desc = scan_node.logical().table_desc();
 
-                    // FIXME: workaround because batch parallel scan on ordered mv may have unorderd
-                    // results. should do a merge sort after parallel scan.
-                    // Tracking issue: <https://github.com/singularity-data/risingwave/issues/3583>
-                    let is_singleton = true;
-                    // let is_singleton = builder.parallelism == 1;
+                    let is_singleton = builder.parallelism == 1;
```
Comment (BugenZhao) [inline src/frontend/src/scheduler/plan_fragmenter.rs:397]: Could you please also make the batch iterator unordered?

https://github.com/singularity-data/risingwave/blob/030a2f80ce77cb1279a3798c20d6bd156b50bcf9/src/storage/src/table/storage_table.rs#L554-L556

**Review by BugenZhao — COMMENTED** [review]
The changes LGTM if we agree on not preserving order on mv scans. 🤣

File: e2e_test/batch/basic/array.slt.part [hunk 2]
```diff
@@ -1,3 +1,5 @@
+control sortmode rowsort
```
Comment (liurenjie1024) [inline e2e_test/batch/basic/array.slt.part:1]: Remove this? This is not querying from mv.

File: e2e_test/batch/basic/boolean.slt.part [hunk 3]
```diff
@@ -1,3 +1,5 @@
+control sortmode rowsort
```
Comment (liurenjie1024) [inline e2e_test/batch/basic/boolean.slt.part:1]: Also remove this?

File: src/frontend/src/scheduler/plan_fragmenter.rs (hunk 1 above)
Comment (liurenjie1024) [inline src/frontend/src/scheduler/plan_fragmenter.rs:397]: How about remove `is_singleton`? Since we no longer need it.

**Review by liurenjie1024 — COMMENTED** [review]
Good job!  Generally LGTM, requires small refine.

File: e2e_test/batch/basic/array.slt.part (hunk 2 above)
Comment (liurenjie1024) [reply e2e_test/batch/basic/array.slt.part:1]: They are same. But as we have discussed, we should not use `rowsort` as default behavior, which is error prone.

**Comment by hengm3467** [comment]
@xxchan Could you please explain the release note part ? It sounds like a limitation. Does this mean that this PR resolves the issue (order by statements return unordered results)? Thanks.

**Comment by xxchan** [comment]
> @xxchan Could you please explain the release note part ? It sounds like a limitation. Does this mean that this PR resolves the issue (order by statements return unordered results)? Thanks.

The problem is described here https://singularity-data.slack.com/archives/C036F5Z3EMD/p1657075038333919

**Pull request merged**